src/pkg/oci/common.go

```diff
@@ -25,69 +23,18 @@ const (
 	MultiOS = "multi"
 )
 
-// Debug does nothing
-func (l *DiscardLogger) Debug(_ string, _ ...any) {}
-
-// Info does nothing
-func (l *DiscardLogger) Info(_ string, _ ...any) {}
-
-// Warn does nothing
-func (l *DiscardLogger) Warn(_ string, _ ...any) {}
-
-// Error does nothing
-func (l *DiscardLogger) Error(_ string, _ ...any) {}
-
-// DiscardLogger is the default if the WithLogger modifier is not used
-// It discards all logs
-type DiscardLogger struct{}
-
-// Logger is an interface built to
-type Logger interface {
-	Debug(msg string, args ...any)
-	Info(msg string, args ...any)
-	Warn(msg string, args ...any)
-	Error(msg string, args ...any)
-}
-
-// Write doesn't do anything but satisfy implementation
-func (DiscardProgressWriter) Write(p []byte) (int, error) {
-	return len(p), nil
-}
-
-// UpdateTitle doesn't do anything but satisfy implementation
-func (DiscardProgressWriter) UpdateTitle(_ string) {}
-
-// DiscardProgressWriter is a ProgressWriter in which all calls succeed without doing anything
-// Use this or nil or if you don't care about writing progress
-type DiscardProgressWriter struct{}
-
-// ProgressWriter wraps io.Writer, but also includes an updateTitle function to give the user
-// additional context on what's going on. Useful in OCI for tracking layers
-type ProgressWriter interface {
-	UpdateTitle(string)
-	io.Writer
-}
-
 // OrasRemote is a wrapper around the Oras remote repository that includes a progress bar for interactive feedback.
 type OrasRemote struct {
 	repo           *remote.Repository
 	root           *Manifest
 	Transport      *helpers.Transport
```

Comment:
have ideas for `Transport` I wanna run by you

Reply:
We discussed making transport private. Instead we just took the Base variable off of transport and composed roundtripper normally. I also made this issue for making progress easier to track in the oci package which could make it easier to make transport private in the future https://github.com/defenseunicorns/zarf/issues/2304